newbank/server/NewBank.java

```diff
@@ -46,14 +46,45 @@ public synchronized String processRequest(CustomerID customer, String request) {
 		if(customers.containsKey(customer.getKey())) {
 			switch(request) {
 			case "SHOWMYACCOUNTS" : return showMyAccounts(customer);
+			case "NEWACCOUNT Savings" : return addSaving(customer);
+			case "NEWACCOUNT Main" : return addMain(customer);
+			case "NEWACCOUNT Checking" : return addChecking(customer);
+
 			default : return "FAIL";
 			}
 		}
 		return "FAIL";
 	}
-	
+
 	private String showMyAccounts(CustomerID customer) {
 		return (customers.get(customer.getKey())).accountsToString();
 	}
 
+	private String addMain(CustomerID customer) {
+		if (showMyAccounts(customer).contains("Main")) {
+			return "FAIL";
+		} else {
+			customers.get(customer.getKey()).addAccount(new Account("Main", 0.0));
+			return "SUCCESS";
```

Comment:
Do we definitely have to restrict to savings/main/checking and not just allow a user to specify an arbitrary name for the account?

Potentially savings/main/checking could be an account type in a later ticket?

Happy for you to decide on this, these functions could be condensed into one with a second argument that contains the account name, for example:

```
addAccount(CustomerID customer, string Name) {
   // same logic but replace the string "Main" with the variable name
}
```

Then happy to approve !

Reply:
I've asked Nadi to check with the customer on whether to allow arbitrary or only these specific names (as well as the question from earlier about listing the possible requests so they don't need to flick back and forth between client and protocol).
Thought I'd put it in like this for now as a sort of filler, and then can generalise if wanted by the customer